pkg/cmd/pulumi/import.go

```diff
@@ -540,8 +543,17 @@ func newImportCmd() *cobra.Command {
 				}
 				importFile = f
 			} else {
-				if len(args) < 3 {
-					return result.Errorf("an inline resource must be specified if no converter or import file is used")
+				if len(args) == 0 {
+					cmd.Help()
+					return result.Errorf("an inline resource must be specified if no converter or import file is used, missing type, name, and id")
+				}
+				if len(args) == 1 {
+					cmd.Help()
+					return result.Errorf("an inline resource must be specified if no converter or import file is used, missing name and id")
+				}
+				if len(args) == 2 {
+					cmd.Help()
+					return result.Errorf("an inline resource must be specified if no converter or import file is used, missing id")
 				}
```

Comment:
This can be simplified:

```go
fields := []string{"type", "name", "id"}
if len(args) > len(fields) {
  return result.Errorf("unexpected arguments: %q", args[len(fields):])
}
missing := fields[:len(args)]
if len(missing) > 0 {
  return result.Errorf("missing fields: %q", missing)
}

// ...
```

(with appropriate tweaks for help messages, etc.)

Reply:
I think I actually prefer the long form. It's a little repetitive but it's simple and it's a clear user facing error. I wouldn't want to shorten this to just "missing fields: id". 

We could have something like "an inline resource must be specified if no converter or import file is used, missing: type, name, id" but then we lose the nice grammar at the end and doing that bit correctly with code seems more complex than helpful.

I will pull the first part of the message out to a common var though, no need to write that out three times.